Add config.py + Remove pysrvx submodule

Fylwind opened the pull request:
- Allow `SECRET_KEY` and `SQLALCHEMY_DATABASE_URI` to be configured using `etc/config.py`.
- Remove the `pysrvx` submodule in favor of installing the dependencies via `pip`.
- Add instructions for how to set up and run the server using [`virtualenv`](https://virtualenv.pypa.io).
- Some other miscellaneous cleanup.

I didn't want to touch `config.yaml` in this PR, but maybe it'd be a good idea also merge `config.py` and `config.yaml` in the future?

Fylwind committed:
Move SECRET_KEY and SQLALCHEMY_DATABASE_URI to config.py

Fylwind committed:
Print errors to stderr and exit with 1 on failure

Fylwind committed:
Remove duplicate entry in .gitignore

Fylwind committed:
Remove pysrvx submodule

Fylwind committed:
Add instructions on how to use virtualenv

staticfox commented:
I believe we're wanting to target at least python3.5. Perhaps the README should be updated to `sudo apt-get install python3-pip python3.5-venv`?

Fylwind commented:
Fixed.

Edit: Actually now that I think about it, I don't think `pip` is necessary at all, so I removed that.  The `virtualenv` command itself appears to be agnostic to the Python version, so I put in an explicit `-p python3` to force Python 3 to be chosen.

staticfox commented:
Thanks!
